src/main/scala/com/databricks/spark/csv/util/TypeCast.scala

```diff
@@ -35,7 +38,8 @@ object TypeCast {
    * @param datum string value
    * @param castType SparkSQL type
    */
-  private[csv] def castTo(datum: String, castType: DataType, nullable: Boolean = true): Any = {
+  private[csv] def castTo(datum: String, castType: DataType, nullable: Boolean = true,
```

Comment:
I don't know MiMa, when we decides the best option, I'll tackle this task if necessary.